### PR TITLE
CNN Why-conv

### PR DESCRIPTION
**Pull request by goldmermaid**

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-zh/PR-581/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/9 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/11 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/12 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/13 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/14 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/15 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/16 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/17 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/18 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

File: chapter_convolutional-neural-networks/why-conv.md [hunk 1]
```diff
@@ -1,115 +1,154 @@
...
 ## 练习
 
-1. 假定卷积内核的大小为 $\Delta = 0$。表明在这种情况下，卷积内核为每组通道独立实现 MLP。
-1. 为什么翻译不变性毕竟不是一个好主意？
-1. 在决定如何处理与图像边界上像素位置相对应的隐藏表示时，我们必须处理哪些问题？
-1. 描述音频的类似卷积层。
-1. 您是否认为卷积图层也可能适用于文本数据？为什么还是为什么不呢？
-1. 证明那是什么
+1. 假设卷积层 :eqref:`eq_conv-layer` 覆盖的局部区域  $\Delta = 0$  。在这种情况下，证明卷积内核为每组通道独立地实现一个完全连接层。
+1. 为什么平移不变性总体上不是一个好主意呢？
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/why-conv.md:148]: 可能不总是一个好主意

Comment (astonzhang) [reply chapter_convolutional-neural-networks/why-conv.md:148]: 原句是 Why might translation invariance not be a good idea after all? 
这个might“可能”的确不能删，否则就不准确了。保证准确性是前提。

或者这样？
“为什么平移不变性可能也不是好主意？”

**Comment by mli** [comment]
Job d2l-zh/PR-581/19 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

File: chapter_convolutional-neural-networks/why-conv.md [hunk 2]
```diff
@@ -1,115 +1,154 @@
 # 从完全连接的层到卷积
 :label:`sec_why-conv`
 
-迄今为止，在我们处理表格数据时，我们迄今讨论的模型仍然是适当的选择。通过表格，我们的意思是数据由与要素对应的示例和列对应的行组成。对于表格数据，我们可能会预计我们所寻求的模式可能涉及要素之间的交互，但我们并不假定任何关于要素交互方式的结构 * 优先级 *。
+我们之前讨论的MLP模型十分适用处理表格数据。这些表格数据，每行分别对应每个样本，每列分别对应每个特征。这些特征可能相互影响，但我们并没有假设这些特征如何交互。然而，对于高维感知数据，这种MLP单一结构模型可能会变得笨拙。
```
[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md:4]: I omit the sentence "Sometimes, we truly lack knowledge to guide the construction of craftier architectures. In these cases, an MLP may be the best that we can do." Do you believe it is necessary to include here? If it is, what is the best way to translate it?

File: chapter_convolutional-neural-networks/why-conv.md [hunk 3]
```diff
@@ -1,115 +1,154 @@
...
 
-有时候，我们确实缺乏指导精巧建筑的建设的知识。在这些情况下，MLP 可能是我们所能做的最好的。然而，对于高维感知数据，这种无结构的网络可能会变得笨拙。
 
-例如，让我们回到我们区分猫和狗的运行例子。假设我们做一个彻底的工作，在数据收集，收集一百万像素照片的注释数据集。这意味着网络的每个输入都有一百万个维度。即使大幅缩小到千个隐藏尺寸，也需要一个以 $10^6 \times 10^3 = 10^9$ 参数为特征的完全连接层。除非我们拥有大量的 GPU、分布式优化的天赋以及极大的耐心，否则学习这个网络的参数可能会被证明是不可行的。
+例如，在之前区分猫和狗的例子中，假设我们收集了一个照片数据集，每张照片具有百万级像素，这意味着MLP的每个输入都有一百万个维度。然而即使MLP只有 $1000$ 个隐藏层，这个神经网络将有 $10^6 \times 10^3 = 10^9$ 个参数。除非我们有大量的GPU，分布式优化训练的经验以及超乎常人的耐心，否则很难学习这个网络。
```
[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md:7]: LOL, any suggestion to translate the last sentence?

File: chapter_convolutional-neural-networks/why-conv.md [hunk 4]
```diff
@@ -1,115 +1,154 @@
...
 
-### 翻译不变
 
-现在让我们援引上面确立的第一个原则：翻译不变性。这意味着输入 $\mathbf{X}$ 的转移应该只会导致隐藏表示 $\mathbf{H}$ 的转移。这只有在 $\mathsf{V}$ 和 $\mathbf{U}$ 实际上并不依赖于 $(i, j)$ 的情况下才可能实现，也就是说，我们有 $[\mathsf{V}]_{i, j, a, b} = [\mathbf{V}]_{a, b}$ 和 $\mathbf{U}$ 是一个常数，比如说 $u$。因此，我们可以简化 $\mathbf{H}$ 的定义：
+$$[\mathbf{H}]_{i, j} = u + \sum_a\sum_b [\mathbf{V}]_{a, b} [\mathbf{X}]_{i+a, j+b}.$$
 
-$$[\mathbf{H}]_{i, j} = u + \sum_a\sum_b [\mathbf{V}]_{a, b}  [\mathbf{X}]_{i+a, j+b}.$$
 
-这是一个 * 卷曲 *！我们正在对位置 $(i, j)$ 附近的像素进行有效的权重，并且系数为 $[\mathbf{V}]_{a, b}$，以获得值 $[\mathbf{H}]_{i, j}$。请注意，$[\mathbf{V}]_{a, b}$ 所需的系数比 $[\mathsf{V}]_{i, j, a, b}$ 少得多，因为它不再依赖于图像中的位置。我们取得了重大进展！
+这就是卷积！我们使用系数 $(i+a, j+b)$ 有效地加权位置 $(i, j)$ 附近的像素以获得隐藏表示 $[\mathbf{H}]_{i, j}$。注意，$[\mathbf{V}]_{a, b}$ 的参数比 $[\mathsf{V}]_{i, j, a, b}$ 少很多，因为前者不再依赖于图像中的位置。太棒了，我们已经通过平移不变性取得了重大进展！
```
[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md:64]: Any recommendation?

File: chapter_convolutional-neural-networks/why-conv.md [hunk 5]
```diff
@@ -1,115 +1,154 @@
...
+
+![沃尔多游戏示例图。](../img/where-wally-walker-books.jpg)
 :width:`400px`
 :label:`img_waldo`
 
-我们现在可以通过列举几个 desiderata 来指导我们设计适用于计算机视觉的神经网络架构，使这些直觉更具体：
 
-1. 在最早的图层中，我们的网络应该对相同的补丁作出类似的响应，而不管它在图像中显示的位置。这个原则被称为 * 翻译不变 *。
-1. 网络的最早图层应该集中于局部区域，而不考虑遥远区域中的图像内容。这是 * 本地 * 原则。最终，可以聚合这些局部制图表达，以便在整个影像级别进行预测。
+现在，我们将如上推断系统化，来设计一下适用于计算机视觉的神经网络体系结构：
```
[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md:31]: Any suggestion for “推断”? How about "想法"？

File: chapter_convolutional-neural-networks/why-conv.md [hunk 6]
```diff
@@ -1,115 +1,154 @@
...
 
 ## 不变性
 
-想象一下，您想要检测图像中的对象。似乎合理的是，无论我们用来识别对象的方法都不应过分关心图像中物体的确切位置。理想情况下，我们的系统应该利用这些知识。猪通常不飞，飞机通常不会游泳。尽管如此，我们仍然应该认识到一只猪是一个出现在图像的顶部。我们可以从儿童游戏 “沃尔多在哪里”（在 :numref:`img_waldo` 中描述）中汲取一些灵感。游戏包括一些混乱的场景与活动爆裂.沃尔多出现在每个地方，通常潜伏在一些不太可能的位置。读者的目标是找到他。尽管他的特色装备，这可能是令人惊讶的困难，由于大量的分心。但是，* 沃尔多看起来像 * 并不取决于沃尔多的位置 *。我们可以使用 Waldo 检测器扫描图像，该检测器可以为每个贴片分配一个分数，指示贴片包含 Waldo 的可能性。CNN 系统化了这个 * 空间不变性 * 的概念，利用它来学习具有较少参数的有用表示。
+想象一下，在一个对象检测任务中，我们不应过分在意图像中物体的确切位置，比如猪通常不在天上飞，飞机通常不游泳。
+我们可以从儿童游戏 ”沃尔多在哪里”（ :numref:`img_waldo` ）中汲取一些灵感。
+这个游戏包括一些混乱的场景，游戏玩家的目标是找到沃尔多,而沃尔多通常潜伏在一些不太可能的位置。
+所以尽管沃尔多的很有特点，在眼花缭乱的场景中找到他也如大海捞针。
+
+由于沃尔多潜藏的地方并不取决于它出现的位置，我们可以使用一个“沃尔多检测器”扫描图像，该检测器将图像分成数个小贴片，并为每个贴片包含沃尔多的可能性打分。而CNN系统化了这个“空间不变性”的概念，用较少参数来学习有用的特征。
```
[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md:23]: Brilliant!

**Comment by mli** [comment]
Job d2l-zh/PR-581/20 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Comment by mli** [comment]
Job d2l-zh/PR-581/21 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

File: chapter_convolutional-neural-networks/why-conv.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (xiaotinghe) [reply chapter_convolutional-neural-networks/why-conv.md]: 有时我们缺乏足够的知识来指导模型的结构设计，此时MLP可能是最好的选择。

File: chapter_convolutional-neural-networks/why-conv.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (xiaotinghe) [reply chapter_convolutional-neural-networks/why-conv.md]: 想要使这个模型学习良好是很难的事情，需要有大量的GPU、分布式优化训练的经验和超乎常人的耐心。

File: chapter_convolutional-neural-networks/why-conv.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (xiaotinghe) [reply chapter_convolutional-neural-networks/why-conv.md]: 现在，我们将上面想法系统总结一下，从而帮助我们设计适合于计算机视觉的神经网络结构：

File: chapter_convolutional-neural-networks/why-conv.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (xiaotinghe) [reply chapter_convolutional-neural-networks/why-conv.md]: 我们使用系数$[\mathbf{V}]_{a, b}$对位置 $(i, j)$ 附近的像素$(i+a, j+b)$进行加权。
太棒了，我们已经通过平移不变性取得了重大进展！其实可以直接不翻译

File: chapter_convolutional-neural-networks/why-conv.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (astonzhang) [reply chapter_convolutional-neural-networks/why-conv.md]: MLP这种不要放英文缩写，用中文全称：多层感知机

[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md]: How about "想要训练这个模型很难，需要有大量的GPU、分布式优化训练的经验和超乎常人的耐心。"?

[the remark this answers is not in the document]
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/why-conv.md]: > MLP这种不要放英文缩写，用中文全称：多层感知机

Fixed!

**Comment by mli** [comment]
Job d2l-zh/PR-581/22 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-581/

**Pull request merged**